crates/hc_sandbox/src/cmds.rs

```diff
@@ -117,33 +117,38 @@ pub struct Existing {
 }
 
 impl Existing {
-    pub fn load(mut self) -> anyhow::Result<Vec<PathBuf>> {
-        let sandboxes = crate::save::load(std::env::current_dir()?)?;
+    /// Determine all sandbox paths to use based on .hc and given options.
+    /// Returns at minimum all paths in self.existing_paths.
+    pub fn load(&self) -> anyhow::Result<Vec<PathBuf>> {
+        let saved_sandboxes = crate::save::load(std::env::current_dir()?)?;
+        let mut provided = self.existing_paths.clone();
         if self.all {
-            // Get all the sandboxes
-            self.existing_paths.extend(sandboxes)
-        } else if self.last && sandboxes.last().is_some() {
-            // Get just the last sandbox
-            self.existing_paths
-                .push(sandboxes.last().cloned().expect("Safe due to check above"));
+            // Add all sandboxes in .hc
+            provided.extend(saved_sandboxes);
+        } else if self.last {
```

Comment:
I'm not sure this is a good change to make. Yes it removes an `expect` call but from the [next version of Rust](https://blog.rust-lang.org/2025/06/26/Rust-1.88.0/#let-chains) it will be possible to write the condition inline anyway.

This introduces some logic that will stand out to me every time I read this which is the pattern of leaving the main logic branch and dropping into sub-logic which has a missing `else`. That will always set off alarm bells for me because it's often a bug.

In this case, it doesn't introduce a bug but it is forcing the code to have both an `else` condition at the bottom of this branch block and then also an `if` check after the block that has a different message but is effectively the same condition.